.github/workflows/build-and-push-containers.yml

```diff
@@ -0,0 +1,47 @@
+name: Build and push containers
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  tags:
+    runs-on: ubuntu-latest
+    outputs:
+      tags: ${{ steps.tags.outputs.tags }}
+    steps:
+      - name: Compile tag list
+        id: tags
+        run: |
+          TAG=${GITHUB_REF/refs\/tags\//}
+          PREFIX=ghcr.io/laminas/automatic-releases
+          MAJOR="${PREFIX}:$(echo ${TAG} | cut -d. -f1)"
+          MINOR="${MAJOR}.$(echo ${TAG} | cut -d. -f2)"
+          PATCH="${PREFIX}:${TAG}"
+          echo "::set-output name=tags::${MAJOR}%0A${MINOR}%0A${PATCH}"
```

Comment:
GHA replaces `%0A` with `\n`, making these line-delimited tags. This creates a list that looks like:

```text
ghcr.io/laminas/automatic-releases:1
ghcr.io/laminas/automatic-releases:1.11
ghcr.io/laminas/automatic-releases:1.11.0
```

which is then passed as the variable "tags" to the next step.

Reply:
This is probably not the best idea to rely on undocumented behavior. Seems like an implementation detail that may behave differently in the future.

However, they have documented to pass as json, if you need something like this:

```yaml
name: build
on: push
jobs:
  job1:
    runs-on: ubuntu-latest
    outputs:
      matrix: ${{ steps.set-matrix.outputs.matrix }}
    steps:
    - id: set-matrix
      run: echo "::set-output name=matrix::{\"include\":[{\"project\":\"foo\",\"config\":\"Debug\"},{\"project\":\"bar\",\"config\":\"Release\"}]}"
  job2:
    needs: job1
    runs-on: ubuntu-latest
    strategy:
      matrix: ${{fromJSON(needs.job1.outputs.matrix)}}
    steps:
    - run: build
```

- https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#example-returning-a-json-object

Reply:
@glensc This is actually the approach recommended by the GHA team, and I discovered it here: https://github.community/t/set-output-truncates-multiline-strings/16852/3

Reply:
I've done some thinking on this, and decided to go with the JSON approach; I'm now doing:

```yaml
with:
  tags: ${{ join(fromJSON(steps.tags.outputs.tags), "\n") }}
```

(as they "tags" value is expected to be multiline).

This should address any concerns about ANSI escape sequences, and ensure we are forwards-compatible with any changes GHA makes.

Reply:
Update: using "," as the separator, as CSV is supported for that value, and the `"\n"` value was causing issues with the GHA expressions.